server/src/main/java/com/projects/pms/filter/LoginFilter.java

```diff
@@ -0,0 +1,130 @@
+import com.projects.pms.request.JwtRequest;
+import com.projects.pms.response.JWTResponse;
+import com.projects.pms.services.EmployeeService;
+import com.projects.pms.util.JwtUtil;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Log4j2
+public class LoginFilter extends AbstractAuthenticationProcessingFilter {
+
+    @Autowired
+    private EmployeeService employeeService;
+
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    JwtUtil jwtUtil;
+
+
+    private final RequestMatcher loginRequestMatcher = new AntPathRequestMatcher("/login",
+            HttpMethod.POST.toString());
+
+    public LoginFilter(String string) {
+        super(string);
+    }
+
+    @Override
+    public Authentication attemptAuthentication( HttpServletRequest request, HttpServletResponse response)
+            throws AuthenticationException, IOException {
+
+        Authentication auth = null;
+
+        System.out.println("attemptAuthentication");
+
+
+
+        if(isLoginRequest(request)){
+            try {
+                JwtRequest creds = new ObjectMapper().readValue(request.getInputStream(), JwtRequest.class);
+                System.out.println("password: "+ creds.getPassword());
+
+                UserDetails userDetails = employeeService.loadUserByUsername(creds.getEmail());
+                System.out.println("loaded user");
+                Authentication authentication = employeeService.authenticate(creds.getEmail(),creds.getPassword());
+                if((!userDetails.getUsername().isEmpty()) && authentication != null){
+                    final String jwt = jwtUtil.generateToken(userDetails);
+                    Cookie cookie = new Cookie("EmpJwt", jwt);
```

Comment:
@SunilSunny698  Have an proper Cookie key instead of EmpJwt

Reply:
Done, renamed to X-AUTH